Brokerages/Bitfinex/BitfinexBrokerage.cs

```diff
@@ -0,0 +1,449 @@
+        {
+            if (this._tickerToken != null)
+            {
+                this._tickerToken.Cancel();
+            }
+        }
+
+        private List<BitfinexOrder> GetOpenBitfinexOrders()
+        {
+            var list = new List<BitfinexOrder>();
+
+            try
+            {
+                var response = _restClient.GetActiveOrders();
+                if (response != null)
+                {
+                    foreach (var item in response)
+                    {
+                        list.Add(new BitfinexOrder
+                        {
+                            Quantity = Convert.ToInt32(decimal.Parse(item.OriginalAmount) * ScaleFactor),
+                            BrokerId = new List<string> { item.Id.ToString() },
+                            Symbol = Symbol,
+                            Time = Time.UnixTimeStampToDateTime(double.Parse(item.Timestamp)),
+                            Price = decimal.Parse(item.Price) / ScaleFactor,
+                            Status = MapOrderStatus(item),
+                            OriginalAmount = decimal.Parse(item.OriginalAmount) * ScaleFactor,
+                            RemainingAmount = decimal.Parse(item.RemainingAmount) * ScaleFactor,
+                            ExecutedAmount = decimal.Parse(item.ExecutedAmount) * ScaleFactor
+                        });
+                    }
+                }
+            }
+            catch (Exception)
+            {
+                throw;
+            }
+            return list;
+        }
+
+        /// <summary>
+        /// Retreive orders from exchange
+        /// </summary>
+        /// <returns></returns>
+        public override List<Order> GetOpenOrders()
+        {
+
+            var list = this.GetOpenBitfinexOrders().Select(o => (Order)o).ToList();
+
+            foreach (var item in list)
+            {
+                if (item.Status != OrderStatus.Canceled && item.Status != OrderStatus.Filled && item.Status != OrderStatus.Invalid)
+                {
+                    var cached = this.CachedOrderIDs.Where(c => c.Value.BrokerId.Contains(item.BrokerId.First()));
+                    if (cached.Count() > 0 && cached.First().Value != null)
+                    {
+                        this.CachedOrderIDs[cached.First().Key] = item;
+                    }
+                }
+            }
+            return list;
+        }
+
+        /// <summary>
+        /// Retreive holdings from exchange
+        /// </summary>
+        /// <returns></returns>
+        public override List<Holding> GetAccountHoldings()
+        {
+
+            var list = new List<Holding>();
+
+            var response = _restClient.GetActivePositions();
+            foreach (var item in response)
+            {
+                var ticker = _restClient.GetPublicTicker(TradingApi.ModelObjects.BtcInfo.PairTypeEnum.btcusd, TradingApi.ModelObjects.BtcInfo.BitfinexUnauthenicatedCallsEnum.pubticker);
```

Comment:
I have a branch (with all planned work completed) ready for another PR with support for all Bitfinex currency pairs.

It makes sense to pull the ticker price once for all positions, but it would mean iterating the holdings twice, once to find out which tickers are needed and then again to deserialize. Would you prefer to leave as if for the time being?